### PR TITLE
Use ESM

### PR DESCRIPTION
**Pull request by wooorm**
### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

* Refactor code style
* Update dependencies, dev-dependencies

Note: I’ll update `mdast-util-to-string` later, which causes a small change in output. And I’ll update to use JSDoc based types in another PR too — turned off dtslint meanwhile

### CODE REVIEW

**Review by ChristianMurphy — APPROVED** [review]

**Review by BarryThePenguin — APPROVED** [review]
Its interesting to see how code can change overtime.. especially with the introduction of esm. Nice!

**Comment by wooorm** [comment]
@BarryThePenguin any particular parts or things that changed you found interesting? Agreed, but curious to hear your thoughts!

File: example.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
   u('heading', {depth: 3}, [u('text', 'Charlie')]),
   u('heading', {depth: 2}, [u('text', 'Delta')])
 ])
 
-var table = toc(tree)
+const table = toc(tree)
 
 // Yields:
-console.log('javascript', require('util').inspect(table, {depth: 3}))
+console.log('javascript', inspect(table, {depth: 3}))
```
Comment (BarryThePenguin) [inline example.js:19]: This is probably more specific to `remark-usage` being used but..

I've never really been a fan of using `require` inline like this. ESM forces static imports to be defined at the beginning of the module.

Sure, there are dynamic imports. However, now that code path needs to be asynchronous

File: lib/contents.js [hunk 2]
```diff
@@ -35,11 +31,11 @@ function contents(map, tight, prefix, ordered) {
...
 // Insert an entry into `parent`.
+// eslint-disable-next-line max-params
 function insert(entry, parent, tight, prefix, ordered) {
-  var siblings = parent.children
-  var tail = siblings[siblings.length - 1]
-  var index = -1
-  var item
+  const siblings = parent.children
+  const tail = siblings[siblings.length - 1]
+  let index = -1
```
Comment (BarryThePenguin) [inline lib/contents.js:38]: Not ESM related.. but the distinction of which `var` became a `let` or `const` is interesting. Seeing the distinction like this is useful 👍🏻

Also, the scope of `item` changing from function scope to block scope 👀

**Pull request merged**